### PR TITLE
feature/banglt/t01-show the total of jobs

### PR DESCRIPTION
**Pull request by letrongbangg**
**Link deploy**
http://banglt-job-board.surge.sh/

**Notes**
- Header component to perform top navigation bar of Job Board
- SearchHeader component perform the total of jobs and a form to search by title of job.
- I used axios.get to get data from mockapi.
- Data fetched (list job) was saved in store of redux.
- The total of jobs was defined by the length of the list job at component SearchHeader
- If there're any error when we get data, it will alert "Get data error" at component SearchHeader

**Result**
1. Get data success
image
2. Get data fail
image

### CODE REVIEW

File: src/components/SearchHeader/styles.scss [hunk 1]
```diff
@@ -0,0 +1,31 @@
+$darkColor: #221f20;
```
Comment (hienntfaba) [inline src/components/SearchHeader/styles.scss:1]: Variable should be placed in a common file

File: src/state/actions/jobActions.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { fetchJobsData } from "../../api/jobApi"
+import { Dispatch } from 'redux'
+import { getJobsAction, getJobsLoading, GET_JOBS, GET_JOBS_ERROR, GET_JOBS_LOADING } from "../constants/jobConstant";
+
+export const getData = (name: string) => async (dispatch: Dispatch<getJobsAction | getJobsLoading>) => {
+  dispatch({ type: GET_JOBS_LOADING, loading: true });
```
Comment (hienntfaba) [inline src/state/actions/jobActions.ts:6]: Don't need `loading: true` because we don't use it in reducer

Comment (hienntfaba) [reply src/state/actions/jobActions.ts]: @letrongbangg , similar to `dispatch({ type: GET_JOBS, payload: jobs, loading: false });` and `dispatch({ type: GET_JOBS_ERROR, payload: "Get data error", loading: false })`

File: src/components/Header/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { render } from '@testing-library/react'
+import { Provider } from 'react-redux';
+import store from '../../state/store';
+import Header from './index'
+
+test("render header bar", async () => {
+  const { container } = render(<Provider store={store}><Header /></Provider>);
+  expect(container).toBeInTheDocument();
```
Comment (hienntfaba) [inline src/components/Header/index.test.tsx:8]: It should be more detailed.

Such as:
- Which text is in the document?
- Which component is in the document?
- ...

File: src/components/SearchHeader/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { render } from "@testing-library/react"
+import { Provider } from "react-redux";
+import SearchHeader from "."
+import store from "../../state/store";
+
+
+describe("SearchHeader", () => {
+  it("should be render number of jobs", () => {
+    const { container } = render(<Provider store={store}><SearchHeader /></Provider>);
+    expect(container).toBeInTheDocument();
```
Comment (hienntfaba) [inline src/components/SearchHeader/index.test.tsx:10]: Same here

File: src/containers/Home/index.test.tsx [hunk 5]
```diff
@@ -1,9 +1,10 @@
...
 import React from "react";
+import { Provider } from "react-redux";
+import store from "../../state/store";
 
 test('renders welcome message', () => {
-  const { getByText } = render(<Home />);
-  const linkElement = getByText(/Welcome to Faba onboarding project - Job board/i);
-  expect(linkElement).toBeInTheDocument();
+  const { container } = render(<Provider store={store}><Home /></Provider>);
+  expect(container).toBeInTheDocument();
```
Comment (hienntfaba) [inline src/containers/Home/index.test.tsx:9]: Same here

File: src/state/actions/jobActions.ts (hunk 2 above)
Comment (letrongbangg) [reply src/state/actions/jobActions.ts]: @hienntfaba file changed

**Pull request merged**